query/src/lib.rs

```diff
@@ -559,6 +559,22 @@ pub struct FindQuery {
     // TODO: in_rules;
 }
 
+impl OrJoin {
+    /// Return true if either the `OrJoin` is `UnifyVars::Implicit`, or if
+    /// every variable mentioned inside the join is also mentioned in the `UnifyVars` list.
+    pub fn is_fully_unified(&self) -> bool {
+        match &self.unify_vars {
+            &UnifyVars::Implicit => true,
+            &UnifyVars::Explicit(ref vars) => {
+                // We know that the join list must be a subset of the vars in the pattern, or
+                // it would have failed validation. That allows us to simply compare counts here.
```

Comment:
Sure, but consider `assert!`-ing the full set equality anyway.  Never know what you'll catch!